### DISCUSSION TITLE
How to use BindingContext when the viewmodel has got DI property in the constructor?

### DISCUSSION THREAD

**Discussion by davecameos** [PrismLibrary/Prism#2298] [Xamarin.Forms]
Hi, 

I registered a service in RegisterTypes using Prism.DryIoc within App.xaml.cs

The view model is something like this:

```
public class MainViewModel
{
    protected IAppInfo AppInfo { get; }
  
    public MainViewModel(IAppInfo appInfo)
    {
        AppInfo = appInfo;
    }
  
    public string Name { get => AppInfo.Name; }
  
    public string Version { get => AppInfo.Version; }
}
```

When I try to bind the context in the corresponding view, the exception appears saying it cannot be used because the view model constructor is not parameterless. 
```
<ContentPage.BindingContext>
    <vm:ViewAPageViewModel />
  </ContentPage.BindingContext>
```
How can I solve this? Thank you

**Comment by brianlagunas** [comment]
Don't manually define the BindingContext, use the ViewModelLocator instead

**Reply by davecameos** [reply]
thank you. I was looking at that. cheers
